### PR TITLE
Fixes #11099: Add a rudder agent diff command

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/11099

### CODE REVIEW

**Comment by amousset** [comment]
Great idea!

File: share/commands/agent-diff [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+# @description show diff between current file and the one before agent modification
+# @man This command will output file change in a diff format
+# @man +
+# @man *Options*:
+# @man +
+# @man *-l*: show diff from the given backup
+# @man +
+# @man *-n*: show diff from the nth backup before the last one
+# @man +
+# @man *-d*: show diff from a given date
```
Comment (amousset) [inline share/commands/agent-diff]: could you explain the expected format?

**Comment by peckpeck** [comment]
Commit modified

File: share/commands/agent-diff [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+# @man +
+# @man *filename*: the file to show diff from
+# @man +
+
+. "${BASEDIR}/../lib/common.sh"
+. "${BASEDIR}/../lib/cfengine_parsing.sh"
+
+# default logfile is last run
+canonify() {
+  echo -n "$1" | tr -c '[A-Za-z0-9_]' '_'
```
Comment (ncharles) [inline share/commands/agent-diff]: echo -n fails on AIX

Comment (ncharles) [inline share/commands/agent-diff]: on AIX
`# echo "12 45 ear?" | tr -c '[A-Za-z0-9_]' '_'`
`
12 45 ear?
`

File: share/commands/agent-diff [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+while getopts "l:n:d:" opt; do
+  case $opt in
+    l)
+      BASE_NAME="${OPTARG}"
+      ;;
+    n)
+      nth="${OPTARG}"
+      ;;
+    d)
+      timestamp=`LANG=C date -d "${OPTARG}" +"%s"`
```
Comment (ncharles) [inline share/commands/agent-diff]: on AIX: date -d 
date: illegal option -- d

Comment (peckpeck) [inline share/commands/agent-diff]: Then the -d option if rudder agent diff is not and cannot be supported for aix
rudder agent log has the same option

**Comment by peckpeck** [comment]
Commit modified

**Pull request merged**